src/observability/watcher.go

```diff
@@ -0,0 +1,80 @@
+package observability
+
+import (
+	"encoding/json"
+	"io"
+
+	"github.com/accuknox/auto-policy-discovery/src/cluster"
+	"github.com/accuknox/auto-policy-discovery/src/types"
+)
+
+func addPodToList(pod types.Pod) {
+	podExist := false
+
+	for _, locpod := range Pods {
+		if locpod.IP == pod.IP && locpod.PodName == pod.PodName && locpod.Namespace == pod.Namespace {
+			podExist = true
+			break
+		}
+	}
+	if !podExist {
+		Pods = append(Pods, types.Pod{
+			Namespace: pod.Namespace,
+			PodName:   pod.PodName,
+			IP:        pod.IP,
+		})
+	}
+}
+
+func updatePodList(pod types.Pod) {
+	for index, locpod := range Pods {
+		if locpod.PodName == pod.PodName && locpod.Namespace == pod.Namespace {
+			Pods[index].IP = pod.IP
+			break
+		}
+	}
+}
+
+// WatchK8sPods Function
+func WatchK8sPods() {
+	for {
+		if resp := cluster.WatchK8sPods(); resp != nil {
+			defer resp.Body.Close()
+
+			decoder := json.NewDecoder(resp.Body)
+			for {
+				event := types.K8sPodEvent{}
+				if err := decoder.Decode(&event); err == io.EOF {
+					break
+				} else if err != nil {
+					break
+				}
+
+				if event.Type != "ADDED" && event.Type != "MODIFIED" && event.Type != "DELETED" {
+					continue
+				}
+
+				pod := types.Pod{
+					Namespace: event.Object.ObjectMeta.Namespace,
+					PodName:   event.Object.ObjectMeta.Name,
+					IP:        event.Object.Status.PodIP,
+				}
+
+				if event.Type == "ADDED" {
+					addPodToList(pod)
+				} else if event.Type == "MODIFIED" {
+					updatePodList(pod)
+				}
```

Comment:
Should we also be deleting the pod's information from the list?

Reply:
Not rquired @Ankurk99 
The reason is that the sumarry logs stays in DB even when pods are removed. 
So when collecting summary output, we need to know the existing pod details as well.